### PR TITLE
Add some utilties to run search queries in parallel in ITs

### PR DESCRIPTION
**Pull request by original-brownbear**
We have loads of tests that assert the same thing about a number of different queries. This introduces some tooling to run some of these spots in parallel.
I only changed a couple of examples in the tests for now, but in general this could be used to save thousands of lines of test code and more importantly, get some coverage on parallel query execution which is covered very little today.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java [hunk 1]
```diff
@@ -375,6 +380,33 @@ public static void assertNoFailuresAndResponse(ActionFuture<SearchResponse> resp
         }
     }
 
+    @SafeVarargs
+    public static <Q extends ActionRequest, R extends ActionResponse> void assertResponses(
```
Comment (javanna) [inline test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java:384]: could you add javadocs?

**Review by javanna — APPROVED** [review]
I like the idea, would be good to use it in more tests!

File: test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java (hunk 1 above)
Comment (original-brownbear) [reply test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java:388]: Done :)

**Comment by original-brownbear** [comment]
Thanks Luca!

**Pull request merged**